ideascube/serveradmin/catalog.py

```diff
@@ -154,8 +154,8 @@ def commit(self):
         for libpath in glob(os.path.join(libdir, '*.xml')):
             zimname = os.path.basename(libpath)[:-4]
 
-            with open(libpath, 'r') as f:
-                et = etree.parse(f)
+            with open(libpath, mode='r', encoding='utf-8') as f:
+                et = etree.fromstring(f.read())
```

Comment:
That's what I did at first. And that would even be the right way to do it, because we shouldn't assume that the XML files are encoded in 'utf-8'.

Unfortunately, the kiwix XML libraries don't include the `encoding` attribute in their XML declaration.

As a result, `parse` would default to the default encoding for the system, which is exactly the problem I'm trying to fix here.

The most correct fix would be this:

```diff
-            with open(libpath, 'r') as f:
+            with open(libpath, 'rb') as f:
```

But that would require Kiwix to provide `library.xml` files with the proper XML declaration:

```xml
<?xml version="1.0" encoding="UTF-8"?>
```

Is that something you can make happen on the Kiwix side?